### PR TITLE
Update to SPK Infra Docs

### LINKED ISSUE #843: Guidance on Infra Deployment Pipeline
**As a:**
Operator

**I want:**
to have complete documentation to create an end-to-end CI/CD infrastructure workflow using `spk` (in Azure DevOps)

**So that:**
I can automate any deployments and/or updates made to my infrastructure according to best practices.

**Describe the solution you'd like:**
A README that will provide guidance around constructing a workflow for deploying infrastructure to production.

- [x] Infrastructure Deployment Pipeline README: 
    - [x] Discuss different approaches to achieving infrastructure deployment securely.
    - [x] Update the original [SPK Infra end-to-end doc](https://github.com/CatalystCode/spk/blob/master/docs/infra/spk-infra-end-to-end.md).
    - [x] Update the original [SPK "under the hood" doc](https://github.com/CatalystCode/spk/blob/master/docs/infra/spk-infra-under-the-hood.md).

**Does this require updates to documentation?:**
Yes

### PR DESCRIPTION
**Pull request by yradsmikham**
Close https://github.com/microsoft/bedrock/issues/843.

### CODE REVIEW

File: docs/cloud-infra-management.md [hunk 1]
```diff
@@ -105,33 +84,34 @@ variables:
...
-Generates a deployment folder of an infrastructure scaffolded project containing
-a `definition.yaml` with a `source`, `template` and `version` to obtain and
-complete the Terraform template files.
+Creates a "generated" deployment folder with the completed Terraform files based
+on definitions provided from a scaffolded project.
 
 It will do the following:
 
 - Check if a provided project folder contains a `definition.yaml`
+- Verify the configiration of parent and leaf definitions.
```
Comment (edaena) [inline docs/cloud-infra-management.md]: configuration

**Review by edaena — APPROVED** [review]

**Review by dennisseah — APPROVED** [review]

**Review by NathanielRose — COMMENTED** [review]
Since we are solely providing documentation for Deployment guidance, should we be touching on mentioning options for source encryption described in https://github.com/microsoft/bedrock/issues/869 or Key Vault versioning options described in https://github.com/microsoft/bedrock/issues/868? If so we can potentially close all 3 tickets with this update of the infra docs.

**Comment by yradsmikham** [comment]
> Since we are solely providing documentation for Deployment guidance, should we be touching on mentioning options for source encryption described in [microsoft/bedrock#869](https://github.com/microsoft/bedrock/issues/869) or Key Vault versioning options described in [microsoft/bedrock#868](https://github.com/microsoft/bedrock/issues/868)? If so we can potentially close all 3 tickets with this update of the infra docs.

@NathanielRose source encryption is low-pri right now and we do not have a timeline as to when to incorporate our findings for that yet. However, we could maybe update the documentation for Generation Pipeline to include knowledge on using Key Vault. I see that https://github.com/microsoft/bedrock/issues/868 is nearly done, with just the versioning and key mapping task left. Should we wait until completion before updating documentation on it? What is the status on it?

@andrebriggs Thoughts?

**Comment by yradsmikham** [comment]
@NathanielRose Also, just for easy record-keeping, I do prefer to have one PR per task. I understand the benefit of hitting multiple birds with one stone, but I think that only makes sense if there is a hard dependency to do so.

**Comment by NathanielRose** [comment]
> > Since we are solely providing documentation for Deployment guidance, should we be touching on mentioning options for source encryption described in [microsoft/bedrock#869](https://github.com/microsoft/bedrock/issues/869) or Key Vault versioning options described in [microsoft/bedrock#868](https://github.com/microsoft/bedrock/issues/868)? If so we can potentially close all 3 tickets with this update of the infra docs.
> 
> @NathanielRose source encryption is low-pri right now and we do not have a timeline as to when to incorporate our findings for that yet. However, we could maybe update the documentation for Generation Pipeline to include knowledge on using Key Vault. I see that [microsoft/bedrock#868](https://github.com/microsoft/bedrock/issues/868) is nearly done, with just the versioning and key mapping task left. Should we wait until completion before updating documentation on it? What is the status on it?

I updated the ticket, it hasn't been worked on. Nor do we use a key vault inside our generation pipeline atm. It's marked as Low Pri and agreed it can be done separately. I wanted to get it back on the radar within this PR to see if it even makes sense adding to the Backlog. @andrebriggs thoughts?

**Comment by andrebriggs** [comment]
@NathanielRose @yradsmikham yes let's keep PRs smaller. In terms of getting the over tickets on the radar can you @ me in them?

**Review by NathanielRose — APPROVED** [review]

**Pull request merged**